scripts/fbi/hate_crime/geo_id_resolver_test.py

```diff
@@ -0,0 +1,166 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import filecmp
+import os
+import tempfile
+import unittest
+from .geo_id_resolver import *
+
+_Test_Cases = [{
+    'input': {
+        'state_abbr': 'NB'
+    },
+    'expected_dcid': 'geoId/31'
+}, {
+    'input': {
+        'state_abbr': 'GM'
+    },
+    'expected_dcid': 'geoId/66'
+}, {
+    'input': {
+        'state_abbr': 'FS'
+    },
+    'expected_dcid': ''
+}, {
+    'input': {
+        'state_abbr': 'AR',
+        'geo_name': 'Sevier',
+        'geo_type': 'County'
+    },
+    'expected_dcid': 'geoId/05133'
+}, {
+    'input': {
+        'state_abbr': 'AR',
+        'geo_name': 'Little Rock',
+        'geo_type': 'City'
+    },
+    'expected_dcid': 'geoId/0541000'
+}]
+
+_County_Test_Cases = [
+    {
+        'input': {
+            'state_abbr': 'AR',
+            'county': 'Governors State University'
+        },
+        'expected_dcid': ''
+    },
+    {
+        'input': {
+            'state_abbr': 'ID',
+            'county': 'Latah'
+        },
+        'expected_dcid': 'geoId/16057'
+    },
+    {
+        'input': {
+            'state_abbr': 'FS',
+            'county': 'Latah'
+        },
+        'expected_dcid': ''
+    },
+    {
+        'input': {
+            'state_abbr': 'MD',
+            'county': 'Montgomery County Police Department'
+        },
+        'expected_dcid': 'geoId/24031'
+    },
+    {
+        'input': {
+            'state_abbr': 'NY',
+            'county': 'Westchester Public Safety'
+        },
+        'expected_dcid': 'geoId/36119'
+    },
+    {
+        'input': {
+            'state_abbr': 'LA',
+            'county': 'Acadia'
+        },
+        'expected_dcid': 'geoId/22001'
+    },
+    {
+        'input': {
+            'state_abbr': 'PA',
+            'county': 'Lancaster '
+        },
+        'expected_dcid': 'geoId/42071'
+    },
+    {
+        'input': {
+            'state_abbr': 'IL',
+            'county': 'DeWitt'
+        },
+        'expected_dcid': 'geoId/17039'
+    },
+    {
+        'input': {
+            'state_abbr': 'ND',
+            'county': 'Lamoure '
+        },
+        'expected_dcid': 'geoId/38045'
+    },
+    {
+        'input': {
+            'state_abbr': 'UT',
+            'county': 'Salt Lake County Unified Police Department'
+        },
+        'expected_dcid': 'geoId/49035'
+    },
+]
+
+_City_Test_Cases = [{
+    'input': {
+        'state_abbr': 'IL',
+        'city': 'Evergreen Park'
```

Comment:
does this resolve even if geo_type  defaults to 'State'?

Would be good to add some test to check failure when geo_type is wrong but entry exists in the maps, such as 
city: "Las vegas"
state_abbr: "CA"
geo_type: "State"
returns ''

Reply:
Ack

Reply:
Currently as per the code, it will resolve the State and ignore the city/county name, as the geo_name is not required for a state (only State_code is sufficient). But I can modify this to return '' instead.

Reply:
Done